### ISSUE TITLE
Sharing: DiscoveryClient does not find running clients/servers

### ISSUE THREAD

**Issue by yacuzo** [Microsoft/MixedRealityToolkit#70]
Hello,
We are using the sharing bit from this repo for Hololens with Unity, but use our own Unity-wiring for the sharing, not the Holotoolkit-Unity scripts. We have used the auto-discovery for months now without problems. Yesterday I updated the binaries to the latest version (recompile this from master and export), and client discovery no longer works. I went back to before pull request #69, built and recompiled and discovery works again.

I tried looking at some relevant cpp files, but none of the ones I find relevant had any significant changes.
I am way out of my depth with C++ and the libraries used here to be able to investigate further.

tldr: Something in the last pull request broke the DiscoveryClient.

[Relevant Holotoolkit-Unity issue](https://github.com/Microsoft/HoloToolkit-Unity/issues/493)

**Comment by NeerajW** [comment]
@yacuzo do you have any specific stack traces or something more to go on?
@matthejo any thoughts here? Looks like quite a few issues being reported after that PR.

**Comment by yacuzo** [comment]
@NeerajW There is no error or exception thrown. It just behaves as if there is no system to be found. There is no output at all. Not even from the debug-build.

It might be the SharingService is not listening on the port, or it might be the DiscoveryClient which is not working properly.

**Comment by yacuzo** [comment]
I used Wireshark to see if any packets were sent: One is sent from client to server and one in sent back. The 
event in the DiscoveryClient C#-wrapper is never raised however. I have a pcap file with the two messages if you are interested. Github won't allow the filetype.

**Comment by yacuzo** [comment]
Can we get this prioritized? It is really tedious to have to change the address, which requires a Unity rebuild, when switching between different devices (unity player, emulator and Hololens).

**Comment by BillMcCrary** [comment]
same experience as @yacuzo no useful log output, no response from service

**Comment by pnolen** [comment]
I've been trying to build the source to see if I can confirm something but i'm running into issues. Can someone take a look at Source/SharingService/SessionServer.cpp? Lines 43 and 440 are commented out. Line 43 specifically is
`//m_profileMgr = new ProfileManagerImpl(m_socketMgr, SystemRole::SessionDiscoveryServerRole);`

This is from the same update where discovery stopped working and I think that's suspicious. :)

For reference:
https://github.com/Microsoft/HoloToolkit/blob/master/Sharing/Src/Source/SharingService/SessionServer.cpp

**Comment by yacuzo** [comment]
I tried uncommenting that line, rebuilding everything and copy to my project. No luck, still no response from server (emulator with service on host machine).

**Comment by StephenHodgson** [comment]
That was such a big PR it's hard to sift though all the files to get a clear idea of what changed.

**Comment by mr0ng** [comment]
I can also confirm that autodiscovery is not working for me. It worked previously.

**Comment by markgrossnickle** [comment]
@NeerajW Bump. Auto discovery is fairly important to client work.  Any chance this is going to be fixed in upcoming changes or should we look into rolling our own solutions?

**Comment by christjt** [comment]
Bump. Any news on this?

**Comment by mr0ng** [comment]
No news, as far as I'm aware. Any guidance on how I might go about troubleshooting / fixing this? I'm happy to put some hours into this, if I know where to start.

**Comment by christjt** [comment]
@mr0ng I honestly have no idea where to even start. The merge that broke it was MASSIVE and my C++ is not good enough to even locate where the problem might be.

**Comment by edwinckc** [comment]
Curious if anyone's found out anything since then? 

Putting the IP into a config file prevents us from having to rebuild in Unity, but we still have to rebuild and deploy in VS. We implemented the HTK keyboard to change IP/port within the scene, but text input was never great to begin with on the HoloLens.

**Comment by StephenHodgson** [comment]
@edwinckc check this out in the meantime: https://github.com/Microsoft/HoloToolkit-Unity/pull/695

**Comment by StephenHodgson** [comment]
The only change I could find that's related to client discovery is [this line that get commented out](https://github.com/Microsoft/HoloToolkit/blob/236bcb725f2f7d99dc827b1981f1ba6c469ea669/Sharing/Src/Source/SharingService/SessionServer.cpp#L43).

**Comment by yacuzo** [comment]
They also removed a lot of dependencies. I suspect they removed one too many RakNet extentions.

**Comment by StephenHodgson** [comment]
Wouldn't that have caused the build to break though?

**Comment by edwinckc** [comment]
Thanks @StephenHodgson , that prefab looks better than the keyboard I've been using, will definitely try to integrate it!

It looks like the profile manager is commented out further down as well ([#440](https://github.com/Microsoft/HoloToolkit/blame/236bcb725f2f7d99dc827b1981f1ba6c469ea669/Sharing/Src/Source/SharingService/SessionServer.cpp#L440)), I wonder if the RakNet changes or something else actually broke something in the profile manager? The profile manager tells the socket manager to accept discovery pings from port 20605, so it makes sense discovery doesn't work with profile manager commented out.

**Comment by StephenHodgson** [comment]
Confirmed tonight that lines 43 & 400 being commented out was the issue. Added those lines back in and rebuilt fixed the issue for me.

**Comment by StephenHodgson** [comment]
Shouldn't be hard, it's just a prefab you add to your scene, and should be fairly simple to setup.

**Comment by sebrk** [comment]
Sonare there any news on this. Has been open for pretty long now?

**Comment by pnolen** [comment]
I thought I got a message about this being fixed by uncommenting lines 43 and 440. Was this not the case?

**Comment by StephenHodgson** [comment]
Was not the case. I thought it bad working but got misinformation from Wireshark.
